File: XPile 3-17/Test/test_yield.js
function* countdown(n) {
    while ((n > 0)) {
        yield n;
        n -= 1;
    }
}

function* doubles(n) {
    let i = 1;
    while ((i <= n)) {
        yield (i * 2);
        i += 1;
    }
}

function main() {
    for (const val of countdown(5)) {
        console.log(val);
    }
    for (const d of doubles(4)) {
        console.log(d);
    }
}


main();

File: XPile 3-17/Test/test_yield.py
from dataclasses import dataclass, field

def countdown(n):
    while (n > 0):
        yield n
        n -= 1

def doubles(n):
    i = 1
    while (i <= n):
        yield (i * 2)
        i += 1

def main():
    for val in countdown(5):
        print(val)
    for d in doubles(4):
        print(d)


if __name__ == "__main__":
    main()
